src/go/k8s/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -493,315 +484,119 @@ func setHelmLabelsAndAnnotations(object client.Object, rp *v1alpha2.Redpanda) {
-		}
-
-		switch kind {
-		case resourceTypeHelmRepository:
-			rp.Status.HelmRepositoryReady = ptr.To(false)
-		case resourceTypeHelmRelease:
-			rp.Status.HelmReleaseReady = ptr.To(false)
-		}
-
-		log.Info(msgNotReady)
-		return false
-	} else if isStatusConditionReady && isStatusReadyNILorFALSE {
-		// here since the condition should be true, we update the value to
-		// be true, and send an event
-		switch kind {
-		case resourceTypeHelmRepository:
-			rp.Status.HelmRepositoryReady = ptr.To(true)
-		case resourceTypeHelmRelease:
-			rp.Status.HelmReleaseReady = ptr.To(true)
-		}
-
-		r.event(rp, rp.Status.LastAttemptedRevision, v1alpha2.EventSeverityInfo, msgReady)
+	if err = json.Unmarshal(values, &partial); err != nil {
+		return nil, err
 	}
 
-	return true
-}
-
-func (r *RedpandaReconciler) reconcileHelmRelease(ctx context.Context, rp *v1alpha2.Redpanda) (*v1alpha2.Redpanda, *helmv2beta2.HelmRelease, error) {
-	var err error
-
-	// Check if HelmRelease exists or create it
-	hr := &helmv2beta2.HelmRelease{}
-
-	// have we recorded a helmRelease, if not assume we have not created it
-	if rp.Status.HelmRelease == "" {
-		// did not find helmRelease, then create it
-		hr, err = r.createHelmRelease(ctx, rp)
-		return rp, hr, err
+	release := helmette.Release{
+		Name:      rp.Name,
+		Namespace: rp.Namespace,
+		// TODO What this Service is used for?
+		//Service:   "",
+		// TODO Should those fields matter in Redpanda, Console and Connectors?
+		//IsUpgrade: false,
+		//IsInstall: false,
 	}
 
-	// if we are not empty, then we assume at some point this existed, let's check
-	key := types.NamespacedName{Namespace: rp.Namespace, Name: rp.Status.GetHelmRelease()}
-	err = r.Client.Get(ctx, key, hr)
+	dot, err := redpanda.Dot(release, partial)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			rp.Status.HelmRelease = ""
-			hr, err = r.createHelmRelease(ctx, rp)
-			return rp, hr, err
-		}
-		// if this is a not found error
-		return rp, hr, fmt.Errorf("failed to get HelmRelease '%s/%s': %w", rp.Namespace, rp.Status.HelmRelease, err)
+		return nil, err
 	}
 
-	// Check if we need to update here
-	hrTemplate, errTemplated := r.createHelmReleaseFromTemplate(ctx, rp)
-	if errTemplated != nil {
-		r.event(rp, rp.Status.LastAttemptedRevision, v1alpha2.EventSeverityError, errTemplated.Error())
-		return rp, hr, errTemplated
-	}
-
-	if r.helmReleaseRequiresUpdate(ctx, hr, hrTemplate) {
-		hr.Spec = hrTemplate.Spec
-		if err = r.Client.Update(ctx, hr); err != nil {
-			r.event(rp, rp.Status.LastAttemptedRevision, v1alpha2.EventSeverityError, err.Error())
-			return rp, hr, err
-		}
-		r.event(rp, rp.Status.LastAttemptedRevision, v1alpha2.EventSeverityInfo, fmt.Sprintf("HelmRelease '%s/%s' updated", rp.Namespace, rp.GetHelmReleaseName()))
-		rp.Status.HelmRelease = rp.GetHelmReleaseName()
-	}
+	dot.KubeConfig = r.kubeConfig
 
-	return rp, hr, nil
-}
+	// Template does not provide a way to pass kubeconfig which lookup needs.
```

Comment:
Yeah, we'll have to fix this. I slammed it in as a hack but it seems it's grown up a bit. We could look into removing all instances of `Lookup` and try to replace it with overrides when possible. Andrew's case for the super user might not be removable though :/